applications/MeshingApplication/python_scripts/mmg_process.py

```diff
@@ -417,12 +419,38 @@ def ExecuteInitializeSolutionStep(self):
                                 self.step = 0  # Reset
                                 self.time = 0.0  # Reset
 
+    def ExecuteFinalize(self):
+        """ This method is executed in order to finalize the simulation and save the refined mesh in a new .mdpa file
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        """
+        self.average_remeshing = self.settings["remesh_post_process"].GetBool()
+        output_file_path = self.settings["output_file_name"].GetString()
+        if self.average_remeshing:
```

Comment:
```suggestion
        self.remesh_at_finalize= self.settings["remesh_at_finalize"].GetBool()
        output_file_path = self.settings["output_file_name"].GetString()
        if self.remesh_at_finalize:
```
There is a mix of variables here. This boolean is controlling if there is a refinement at the end of the simulation (ExecuteFinalize()), so the variable `self.average_remeshing` is not accurate.